### PR TITLE
[security] update `sinatra` dev dependency to resolve dependabot alert

### PR DESCRIPTION
**Pull request by lacostej**
See https://github.com/fastlane/fastlane/security/dependabot

### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [ ] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.
- [ ] I've added or updated relevant unit tests.

### CODE REVIEW

**Review by AliSoftware — CHANGES_REQUESTED** [review]

File: fastlane.gemspec [hunk 1]
```diff
@@ -129,7 +129,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency('rubocop-performance')
   spec.add_development_dependency('rubocop-require_tools')
   spec.add_development_dependency('rubocop', Fastlane::RUBOCOP_REQUIREMENT)
-  spec.add_development_dependency('sinatra', '~> 2.0.8') # Used for mock servers
+  spec.add_development_dependency('sinatra', '>= 2.0.8') # Used for mock servers
```
[the remark this answers is not in the document]
Comment (lacostej) [reply fastlane.gemspec:132]: Well, sinatra is used for the tests and not a dependency to users of fastlane, so it's a rather internal library.

As we always lock the versions in the lock file, if there's a breakage in the future, if will only happen if a dev forces a new version in the lock file in a PR. The PR checks should detect the issue and we could lock down the range or update the code then. WDYT?

[the remark this answers is not in the document]
Comment (AliSoftware) [reply fastlane.gemspec:132]: Well the point of using `~>` is so that even if one runs `bundle outdated` or `bundle update` this will guarantee not the break the API, so that's why I'm always keen to using semver and ~> to prevent accidental breakage anyway.

That being said I see your point of this just being a dev dependency and having less impact if it breaks; but still?

Maybe we need a third opinion 😛
cc @rogerluan @joshdholtz ?

[the remark this answers is not in the document]
Comment (rogerluan) [reply fastlane.gemspec:132]: Does `~>=` exist? 🤔 

If this is a dev-only gem (and I checked that it indeed is only being `require`'d in specs), then I'd be in favor of moving this to `Gemfile` instead.

I'm in favor of us adhering to the best practice of having development dependencies for a gem being specified in the `Gemfile`, rather than in the `gemspec` using `add_development_dependency`. This is even covered by RuboCop: https://docs.rubocop.org/rubocop/cops_gemspec.html#gemspecdevelopmentdependencies

Once in the Gemfile, I'm ok with doing _nearly_ whatever we want with it 😄 generally speaking the point of using `~>` is so that everyone in the team uses the same major version (in this case, v2), so if we want to allow some people to use v3, I'd rather for us to be intentional and upgrade everyone to v3 (properly addressing any migration points we might need/have).

tl;dr: I think `~>` is better (for us, maintainers — users won't be bothered), and we can bump to v3 separately.

What do you guys think?

[the remark this answers is not in the document]
Comment (lacostej) [reply fastlane.gemspec:132]: @rogerluan what did you mean with `We declare development dependencies in the Gemspec file because we use dynamic values in its template`, message introduced in the gemspec in #21465.

[the remark this answers is not in the document]
Comment (rogerluan) [reply fastlane.gemspec:132]: Ooh I had completely forgotten about that @lacostej 😄 

This code in particular disables the aforementioned rubocop rule:

```ruby
config['Gemspec/DevelopmentDependencies'] = {
   'Enabled' => false
 }
```

For RuboCop config files (templates). They are when you generate a new plugin, using fastlane's plugin generator

So the comment only applies to `fastlane/lib/fastlane/plugins/template/%gem_name%.gemspec.erb` AFAIK… which is why I didn't move `fastlane/lib/fastlane/plugins/template/%gem_name%.gemspec.erb`'s `add_development_dependency` to `fastlane/lib/fastlane/plugins/template/Gemfile`. It's not that it can't be moved, but it'd have required more changes at the time, in a already big PR, so I decided to not increase the scope to that point. We should eventually migrate them, though 👀

[the remark this answers is not in the document]
Comment (lacostej) [reply fastlane.gemspec:132]: @rogerluan Do you mean that if we rewrite the plugin generator to move its development dependencies into its Gemfile, we could also do the same with fastlane's gemspec/Gemfile?

[the remark this answers is not in the document]
Comment (rogerluan) [reply fastlane.gemspec:132]: @lacostej We can do that to fastlane's independently. Doing that to the plugin template would be a nice-to-have on top of that 😃

[the remark this answers is not in the document]
Comment (lacostej) [reply fastlane.gemspec]: OK. For now I've focused on bumping the dependency to get rid of the security warning. PR updated to satisfy @AliSoftware 

We can create 2 different PRs for fixing the plugin template and fastlane itself later on.

**Review by rogerluan — APPROVED** [review]
💪

File: fastlane.gemspec (hunk 1 above)
[the remark this answers is not in the document]
Comment (rogerluan) [reply fastlane.gemspec]: Here are the 2 new PRs to address those issues:

- https://github.com/fastlane/fastlane/pull/21725
- https://github.com/fastlane/fastlane/pull/21726

**Review by AliSoftware — APPROVED** [review]
